Application/EdFi.Ods.Api/Security/AuthorizationStrategies/Relationships/RelationshipsAuthorizationStrategyBase.cs

```diff
@@ -116,20 +116,26 @@ string GetNonRoleNamedProperty(string propertyName)
             {
                 //Look for the corresponding non - role named property on the model using its DefiningConcreteProperty
 
-                var schema = authorizationContext.Type.GetCustomAttribute<SchemaAttribute>(false)?.Schema;
-                var expectedEntityFullName = new FullName(schema, authorizationContext.Type.Name);
-
-                if (!_domainModel.EntityByFullName.TryGetValue(expectedEntityFullName, out var entity))
+                return _nonRoleNamedByRoleNamedPropertyName.GetOrAdd(propertyName, _ =>
                 {
-                    throw new Exception($"Unable to locate entity '{expectedEntityFullName}' in the model.");
-                }
+                    var schema = authorizationContext.Type.GetCustomAttribute<SchemaAttribute>(false)?.Schema ??
+                                 throw new Exception(
+                                     $"The {nameof(SchemaAttribute)} is required for the entity '{authorizationContext.Type.FullName}'.");
 
-                if (!entity.PropertyByName.TryGetValue(propertyName, out var property))
-                {
-                    throw new Exception($"Property {propertyName} not found in entity '{expectedEntityFullName}'.");
-                }
+                    var expectedEntityFullName = new FullName(schema, authorizationContext.Type.Name);
+
+                    if (!_domainModel.EntityByFullName.TryGetValue(expectedEntityFullName, out var entity))
+                    {
+                        throw new Exception($"Unable to locate entity '{expectedEntityFullName}' in the model.");
+                    }
 
-                return property.DefiningConcreteProperty.PropertyName;
+                    if (!entity.PropertyByName.TryGetValue(propertyName, out var property))
+                    {
+                        throw new Exception($"Property {propertyName} not found in entity '{expectedEntityFullName}'.");
```

Comment:
Add single quotes around `{propertyName}` in the message.

Reply:
Fixed it